secant/Features/WalletEventsFlow/Views/TransactionDetailView.swift

```diff
@@ -1,11 +1,146 @@
 import SwiftUI
+import ComposableArchitecture
 
 struct TransactionDetailView: View {
     var transaction: TransactionState
+    var viewStore: WalletEventsFlowViewStore
+    
     var body: some View {
-        Text(String(dumping: transaction))
+        ScrollView {
+            HStack {
+                Text("\(transaction.date.asHumanReadable())")
+                Spacer()
+                Text("HEIGHT \(heightText)")
+            }
             .padding()
-            .navigationTitle("Transaction: \(transaction.id)")
+
+            Text("\(amountPrefixText) \(transaction.zecAmount.decimalString()) ZEC")
+                .transactionDetailRow()
+            
+            Text("fee \(transaction.fee.decimalString()) ZEC")
+                .transactionDetailRow()
+
+            Text("total amount \(transaction.totalAmount.decimalString()) ZEC")
+                .transactionDetailRow()
+
+            Button {
+                viewStore.send(.copyToPastboard(transaction.address))
+            } label: {
+                Text("\(addressPrefixText) \(transaction.address)")
+                    .lineLimit(1)
+                    .truncationMode(.middle)
+                    .transactionDetailRow()
+            }
+
+            if let memo = transaction.memo {
+                Button {
+                    viewStore.send(.copyToPastboard(memo))
+                } label: {
+                    VStack {
+                        Text("\(memo)")
+                            .multilineTextAlignment(.leading)
+                        
+                        HStack {
+                            Text("reply-to address included")
+                            Spacer()
+                            Button {
+                                viewStore.send(.replyTo(transaction.address))
+                            } label: {
+                                Text("reply now")
+                                    .padding(5)
+                                    .overlay(
+                                        RoundedRectangle(cornerRadius: 6)
+                                            .stroke(Asset.Colors.Text.transactionDetailText.color, lineWidth: 1)
+                                    )
+                            }
+                        }
+                    }
+                    .transactionDetailRow()
+                }
+            }
+
+            HStack {
+                Text("Confirmed")
+                Spacer()
+                Text("\(transaction.confirmations) times")
+            }
+            .transactionDetailRow()
+
+            Spacer()
+            
+            Button {
+                viewStore.send(.copyToPastboard(transaction.id))
+            } label: {
+                Text("txn:\(transaction.id)")
```

Comment:
```suggestion
                Text("txn: \(transaction.id)")
```

Reply:
FYI: Initially I purposely removed the space, otherwise the layout breaks into 3 lines.. We will need to implement it smartly when finishing the UI. 

I put the space back so it's visible what we need to handle.